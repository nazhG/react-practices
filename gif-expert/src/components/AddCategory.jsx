import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {

    const [input, setInput] = useState('');

    const handleInput = (e) => {
        setInput(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(input);
        
        if (input.trim().length < 2) {
            return;
        }

        onNewCategory(input);
        setInput('');
    }

    return (
        <form onSubmit={handleSubmit} name="form">
            <input 
                type="text" 
                name="category" 
                id="cat" 
                value={input} 
                onChange={handleInput}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}