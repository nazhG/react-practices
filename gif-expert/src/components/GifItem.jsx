import React from 'react'
import PropTypes from 'prop-types';

export const GifItem = ({id, url, title }) => {
  return (
    <img 
      key={id} 
      src={url} 
      alt={title} 
      className="gif-item"
    />
  )
}

//npm i prop-types
GifItem.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}