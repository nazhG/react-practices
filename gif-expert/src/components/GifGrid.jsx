import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
    
    const { images, loading, error } = useFetchGifs(category);

    return (
        <>
            <h1>{category}</h1>
            <div className="gif-grid">
                {/* {images.map(({id, images:{downsized_medium:{url}}, title}) => ( */}
                {images.map(image => (
                    /* !loading && (<h2>Cargando ...</h2>) */
                    loading?
                    (<h2>Cargando ...</h2>):
                    GifItem({ ...image })
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
