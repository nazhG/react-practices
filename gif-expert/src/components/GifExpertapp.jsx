import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpertapp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const onAddCategory = (cat) => {
    if (categories.indexOf(cat) === -1) {
      setCategories([cat, ...categories]);
    }
  };

  return (
    <>
      <h1>GifExpertapp</h1>

      <AddCategory onNewCategory={(cat) => onAddCategory(cat)} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
