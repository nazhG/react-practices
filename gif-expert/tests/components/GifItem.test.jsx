import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';

    test('debe de hacer mach con el snapshot', () => {
        const { container } = render( <GifItem title={title} url={url} id="1" /> );
        expect(container).toMatchSnapshot();
    });
    
})