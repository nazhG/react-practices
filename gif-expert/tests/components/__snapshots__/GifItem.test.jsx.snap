// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem debe de hacer mach con el snapshot 1`] = `
<div>
  <img
    alt="Un titulo"
    class="gif-item"
    src="https://localhost/algo.jpg"
  />
</div>
`;
